pkg/kubectl/stop.go

```diff
@@ -361,21 +361,17 @@ func (reaper *DeploymentReaper) Stop(namespace, name string, timeout time.Durati
 	replicaSets := reaper.Extensions().ReplicaSets(namespace)
 	rsReaper, _ := ReaperFor(extensions.Kind("ReplicaSet"), reaper)
 
-	deployment, err := deployments.Get(name)
-	if err != nil {
-		return err
-	}
-
-	// set deployment's history and scale to 0
-	// TODO replace with patch when available: https://github.com/kubernetes/kubernetes/issues/20527
-	zero := 0
-	deployment.Spec.RevisionHistoryLimit = &zero
-	deployment.Spec.Replicas = 0
-	// TODO: un-pausing should not be necessary, remove when this is fixed:
-	// https://github.com/kubernetes/kubernetes/issues/20966
-	// Instead deployment should be Paused at this point and not at next TODO.
-	deployment.Spec.Paused = false
-	deployment, err = deployments.Update(deployment)
+	deployment, err := reaper.updateDeploymentWithRetries(namespace, name, func(d *extensions.Deployment) {
+		// set deployment's history and scale to 0
+		// TODO replace with patch when available: https://github.com/kubernetes/kubernetes/issues/20527
+		zero := 0
+		d.Spec.RevisionHistoryLimit = &zero
```

Comment:
There is a `IntPtr()` helper functions in pkg/util.go: https://github.com/kubernetes/kubernetes/blob/master/pkg/util/util.go#L120